File: app/src/main/java/com/hiddenramblings/tagmo/AmiiTool.java
package com.hiddenramblings.tagmo;

public class AmiiTool {
    static {
        System.loadLibrary("amiitool");
    }

    public native int setKeysFixed(byte[] data, int length);

    public native int setKeysUnfixed(byte[] data, int length);

    public native int unpack(byte[] tag, int tagLength, byte[] unpackedTag, int unpackedTagLength);

    public native int pack(byte[] tag, int tagLength, byte[] unpackedTag, int unpackedTagLength);
}

File: app/src/main/java/com/hiddenramblings/tagmo/AmiiTool.kt
package com.hiddenramblings.tagmo

class AmiiTool {
    external fun setKeysFixed(data: ByteArray?, length: Int): Int
    external fun setKeysUnfixed(data: ByteArray?, length: Int): Int
    external fun unpack(
        tag: ByteArray?,
        tagLength: Int,
        unpackedTag: ByteArray?,
        unpackedTagLength: Int
    ): Int

    external fun pack(
        tag: ByteArray?,
        tagLength: Int,
        unpackedTag: ByteArray?,
        unpackedTagLength: Int
    ): Int

    companion object {
        init {
            System.loadLibrary("amiitool")
        }
    }
}